src/plots/cartesian/axis_defaults.js

```diff
@@ -189,3 +210,18 @@ function rangebreaksDefaults(itemIn, itemOut, containerOut) {
         }
     }
 }
+
+var weekSTR = [
+    'sun',
+    'mon',
+    'tue',
+    'wed',
+    'thu',
+    'fri',
+    'sat'
+];
+
+function indexOfDay(v) {
+    var str = String(v).substr(0, 3).toLowerCase();
+    return weekSTR.indexOf(str);
```

Comment:
This could be sped up a bit with an object lookup rather than `indexOf`
```js
var dayStrToNum = {
    sun: 0,
    mon: 1,
    ...
}

return dayStrToNum[str];
```

Reply:
and possibly bail out early if `v` isn't already a string, rather than converting to a string only to discard it later.

Reply:
Good calls! Done in https://github.com/plotly/plotly.js/pull/4661/commits/db2c3a3af9018477def1a6abb22b88b70c5081ab.
Curious to know if there is a good article on this?
At the moment, there are quite a number of places that `indexOf !== -1` is applied, so please feel free to open an issue for that.
Thanks.

Reply:
Not sure about an article... you find a lot of people saying "hash lookup is O(1) but indexOf is O(n)" but that's as far as it usually goes.

Here's a jsperf showing the general trends: https://jsperf.com/array-indexof-vs-object-key-lookup2/12 - basically, object lookups hardly scale with the number of keys but `indexOf` scales linearly. So `indexOf` can be a nice way to simplify `if (a===b || a===c || a===d)` -> `if ([b,c,d].indexOf(a) !== -1)` - but more than a few items and it's better to turn the problem into an object lookup.

Also of course object lookups only support strings... looking forward to when we stop supporting IE and can use [`Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set)!

Reply:
@alexcjohnson thank you!!